docs_nnx/guides/jax_and_nnx_transforms.rst

```diff
@@ -18,27 +17,34 @@ First, let's set up imports and generate some dummy data:
   x = jax.random.normal(jax.random.key(0), (1, 2))
   y = jax.random.normal(jax.random.key(1), (1, 3))
 
-Differences between NNX and JAX transformations
-***********************************************
+Differences
+***********
+
+Flax NNX transformations can transform functions that are not pure and make mutations and
+side-effects:
+- Flax NNX transforms enable you to transform functions that take in Flax NNX graph objects as
+arguments - such as ``nnx.Module``, ``nnx.Rngs``, ``nnx.Optimizer``, and so on - even those whose state
+will be mutated.
+- In comparison, this kind of functions aren't recognized in JAX transformations.
+
+The Flax NNX `Functional API <https://flax.readthedocs.io/en/latest/nnx/nnx_basics.html#the-functional-api>`_
+provides a way to convert graph structures to `pytrees <https://jax.readthedocs.io/en/latest/working-with-pytrees.html>`__
+and back. By doing this at every function boundary you can effectively use graph structures with any
+JAX transform, and propagate state updates in a way consistent with functional purity.
 
-The primary difference between Flax NNX and JAX transformations is that Flax NNX transformations allow you to
-transform functions that take in Flax NNX graph objects as arguments (`Module`, `Rngs`, `Optimizer`, etc),
-even those whose state will be mutated, whereas they aren't recognized in JAX transformations.
-Therefore Flax NNX transformations can transform functions that are not pure and make mutations and
-side-effects.
+Flax NNX custom transforms, such as ``nnx.jit`` and ``nnx.grad``, simply remove the boilerplate, and
+as a result the code looks stateful.
 
-Flax NNX's `Functional API <https://flax.readthedocs.io/en/latest/nnx/nnx_basics.html#the-functional-api>`_
-provides a way to convert graph structures to pytrees and back. By doing this at every function
-boundary you can effectively use graph structures with any JAX transform and propagate state updates
-in a way consistent with functional purity. Flax NNX custom transforms such as ``nnx.jit`` and ``nnx.grad``
-simply remove the boilerplate, as a result the code looks stateful.
+Below is an example of using the ``nnx.jit`` and ``nnx.grad`` transforms compared to the
+the code that uses ``jax.jit`` and ``jax.grad`` transforms.
 
-Below is an example of using the ``nnx.jit`` and ``nnx.grad`` transformations compared to using the
-``jax.jit`` and ``jax.grad`` transformations. Notice the function signature of Flax NNX-transformed
-functions can accept the ``nnx.Linear`` module directly and can make stateful updates to the module,
-whereas the function signature of JAX-transformed functions can only accept the pytree-registered
-``State`` and ``GraphDef`` objects and must return an updated copy of them to maintain the purity of
-the transformed function.
+Notice that:
+
+- The function signature of Flax NNX-transformed functions can accept the ``nnx.Linear``
+  ``nnx.Module`` directly and can make stateful updates to the ``Module``.
```

Comment:
"can accept the ``nnx.Linear`` ``nnx.Module`` instances directly"

Reply:
Thank you!

> "can accept ... instances directly and make stateful updates to the Module"

(removed the second "can")

```
- The function signature of Flax NNX-transformed functions can accept the ``nnx.Linear``
  ``nnx.Module`` instances directly and make stateful updates to the ``Module``.
```